program-runtime/src/loaded_programs.rs

```diff
@@ -914,6 +952,39 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
         self.unload_program_entries(sorted_candidates.iter().take(num_to_unload));
     }
 
+    /// Evicts programs using 2's random selection, choosing the least used program out of the two entries.
+    /// The eviction is performed enough number of times to reduce the cache usage to the given percentage.
+    pub fn evict_using_2s_random_selection(&mut self, shrink_to: PercentageInteger, now: Slot) {
```

Comment:
@behzadnouri mind giving this method a once over?

Reply:
@Lichtso, @alessandrod?

Reply:
the index selection code lgtm